doc/explanation/mental-model.rst

```diff
@@ -0,0 +1,175 @@
+- (but it is attached to ``runtest``)
+
+How Dune Interprets Rules
+-------------------------
+
+Now that we have seen how Dune sets up rules for a project, let's see how it
+executes a ``dune build`` command.
+
+Dune can build targets: either files, or aliases. ``dune build tool.exe`` will
+build ``tool.exe``, while ``dune build @runtest`` will build the ``runtest``
+alias (``dune runtest`` is a shortcut for ``dune build @runtest``).
+
+To build the target *t*, Dune will first check is *t* is in the source tree. In
+that case, it will copy it to the ``_build`` directory. Otherwise, it will
+check if it is the target of a rule *r*. In that case, it will build all
+dependencies of *r* and if any dependency has changed, it will run the action
+of *t*. (Dune will raise an error in other cases: if *t* is both in the source
+tree and the target of a rule, or if it is neither)
+
+In the case of our example, if we call ``dune runtest``, Dune will consider all
+rules attached to the ``runtest`` alias. This is just the integration test
+rule. It needs to build its dependencies, ``tool.exe`` and ``testdata.txt``.
+The latter is present in the source tree, so it just needs to be copied.
+However, ``tool.exe`` is the target of the linking rule defined by the
+``(executable)`` stanza. This rule requires ``main.cmx`` and ``config.cmx``.
+``main.cmx`` is the target of the compilation rule for the ``Main`` module,
+which depends on ``main.ml``. This file is in the source tree, so let's copy it
+under ``_build``. This rule has all its dependencies available, so we can run
+its action, which writes ``main.cmx``. Getting back to the dependencies of
+``tool.exe``, ``config.cmx`` is the target of the linking rule of the
+``Config`` module. This rule has ``config.ml`` has a dependency. This file is
+itself the target of the configuration module rule, which lists ``config.json``
+and ``convert/json2ml.exe``. The first is available in the source tree and to
+simplify, let's assume that the second one has been built. This action has all
+its dependencies available, so we can execute its action to produce its target,
+``config.ml``. Now the module compilation rule for ``Config`` can be executed,
+producing ``config.cmx``; and in turn the linking rule can be executed,
+producing ``tool.exe``. Finally, ``tool.exe`` can be executed with
+``testdata.txt`` as its argument.
+
+In a nutshell: we recursively copied all the dependencies of the test rule, and
+executed the rules in the correct order.
+
+This is a "cold build", where nothing was previously built. But note that if we
+change only part of the project (say the ``main.ml`` file), only a small number
+of rules will be evaluated, the ones that depend on ``main.ml``.
+
+Conclusion
+----------
+
+Dune's underlying model is based on rules. Stanzas are high-level constructs
+that can generate multiple rules, that are not always visible.
+
+To build a target, Dune looks for the rule that produces that target and makes
```

Comment:
```suggestion
To build a target or alias, Dune looks for the rule that produces that target and makes
```
Not sure this change is necessary, but since the `(target ...)` syntax is used for file targets and `(alias ...)` is used for aliases, using the word "target" here might suggest that the behaviour when building an "alias" is different.

Reply:
Oh actually this paragraph reads as though it's only about file targets, in which case I'd say leave it as such and add an additional sentence summarizing how dune builds aliases.